tests/cmdline/commands/test_status.py

```diff
@@ -98,3 +98,32 @@ def storage_cls(*args, **kwargs):
     result = run_cli_command(cmd_status.verdi_status, raises=True, use_subprocess=False)
     assert 'Storage is corrupted' in result.output
     assert result.exit_code is ExitCode.CRITICAL
+
+
+def test_sqlite_version(run_cli_command, monkeypatch):
+    """Test `verdi status` when the storage is found to be corrupt (e.g. non-matching repository UUIDs)."""
+
+    profile = get_profile()
+    storage_backend = profile._attributes['storage']['backend']
+    # This should be True, only if `pytest -m 'presto'` is used.
+    # and that is essential to guarantee the funtionality of the code!
+    if storage_backend in ['core.sqlite_dos', 'core.sqlite_zip']:
```

Comment:
Will this ever be the case though? If you really want to test this behavior in `verdi status`, I think you should _force_ creating a profile with the relevant storage backend and then test it.

Reply:
So this works as expected only in `tests-presto`. It's being triggered only via `pytest -m 'presto'`.

I've tried your suggestion to create a profile for each storage backend, but there is a strange problem there, which I don't understand. 
Suppose I do:
```python
@pytest.mark.parametrize('entry_point', ('core.sqlite_zip', 'core.sqlite_dos', 'core.psql_dos'))
def test_sqlite_version(run_cli_command, monkeypatch, entry_point, tmp_path, config_psql_dos):
    if entry_point == 'core.sqlite_zip':
        tmp_path = tmp_path / 'archive.aiida'
        create_archive([], filename=tmp_path)

    if entry_point == 'core.psql_dos':
        options = []
        for key, value in config_psql_dos().items():
            options.append(f'--{key.replace("_", "-")}')
            options.append(str(value))
    else:
        options = ['--filepath', str(tmp_path)]

    profile_name = f'temp-profile{entry_point}'
    options = [entry_point, '-n', '--profile-name', profile_name, '--email', 'email@host', '--set-as-default', *options]
    result = run_cli_command(cmd_profile.profile_setup, options, use_subprocess=True)
    assert f'Created new profile `{profile_name}`.' in result.output

    if entry_point in ['core.sqlite_dos', 'core.sqlite_zip']:
->        result = run_cli_command(cmd_status.verdi_status, use_subprocess=True, raises=True)
```

This last line returns wrong results for some other profile, even if I `set-as-default` explicitly:
AssertionError:  ✔ version:     AiiDA v2.6.2.post0
   ✔ config:      /tmp/pytest-of-khosra_a/pytest-13/c048339d34c915b04f9e61ba1aa470df0/.aiida
   ✔ profile:     f4366537365f007119dd90576b2ab04d
   ✔ storage:     Storage for 'f4366537365f007119dd90576b2ab04d' [open] @ postgresql+psycopg://guest:***@localhost:54571/61eeacfc-85be-458a-93de-c198106f23c5 / DiskObjectStoreRepository: 43c774c5c42a44df98360169b9f08ea2 | /tmp/pytest-of-khosra_a/pytest-13/repository0/container
  Warning: RabbitMQ v3.12.1 is not supported and will cause unexpected problems!
  Warning: It can cause long-running workflows to crash and jobs to be submitted multiple times.
  Warning: See https://github.com/aiidateam/aiida-core/wiki/RabbitMQ-version-to-use for details.
   ✔ broker:      RabbitMQ v3.12.1 @ amqp://guest:guest@127.0.0.1:5672?heartbeat=600
   ⏺ daemon:      The daemon is not running.
  
While if you do this:
```python
(Pdb) import os
(Pdb) os.system("verdi status")
```
returns the correct one:

 ✔ version:     AiiDA v2.6.2.post0
 ✔ config:      /tmp/pytest-of-khosra_a/pytest-10/fc80b65e071f67ef50d89cc715645faa0/.aiida
 ✔ profile:     temp-profilecore.sqlite_dos
 ✔ storage:     SqliteDosStorage[/tmp/pytest-of-khosra_a/pytest-10/test_sqlite_version_core_sqlit0]: open,
Warning: RabbitMQ v3.12.1 is not supported and will cause unexpected problems!
Warning: It can cause long-running workflows to crash and jobs to be submitted multiple times.
Warning: See https://github.com/aiidateam/aiida-core/wiki/RabbitMQ-version-to-use for details.
 ✔ broker:      RabbitMQ v3.12.1 @ amqp://guest:guest@127.0.0.1:5672?heartbeat=600
 ⏺ daemon:      The daemon is not running.

Reply:
Not quite sure why it is not picking up the new profile. But anyway, this is not the way you probably want to go about this. You are now creating a profile in the test configuration. This can affect other tests. You probably want to create a new temporary isolated config, create the profile in that one, and just run the cli command in memory. See this test as an example of how to create the profile:
https://github.com/aiidateam/aiida-core/blob/98ffc331d154cc7717861fde6c908ec510003926/tests/manage/configuration/test_configuration.py#L10

And then just run `run_cli_command(use_subprocess=False)`. This should be a lot faster and should keep things nicely isolated.

Reply:
@sphuber  I follow your latest suggestion to make an isolated profile. 
Unfortunately, the issue that I posted in my previous comment, still persist. And I have no clue why.
The profile that I make (either `'core.sqlite_dos'` or `'core.sqlite_zip'` ) are not picked up by `run_cli_command(cmd_status.verdi_status, use_subprocess=False)` when tests are run with `psql` database backend. 

I don't know what to do with this... 
Is it ok if we bound this test to run only when pytest is running with `sqlite_dos` storage backend?